### DISCUSSION TITLE
How to configure listen address?

### DISCUSSION THREAD

**Discussion by gosssamer** [mcguirepr89/BirdNET-Pi#1020] [General]
Hi, I've just installed birdnet successfully on my raspberry using the "comprehensive install guide" and can access it successfully using ssh. However, there's nothing listening on ipv4 port 80:

```
$ netstat -ntap|grep LISTEN
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN      408/gotty           
tcp        0      0 127.0.0.1:2019          0.0.0.0:*               LISTEN      -                   
tcp        0      0 0.0.0.0:8000            0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:8888          0.0.0.0:*               LISTEN      -                   
tcp        0      0 0.0.0.0:21              0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      413/python3         
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:8501          0.0.0.0:*               LISTEN      415/python3         
tcp6       0      0 ::1:8501                :::*                    LISTEN      415/python3         
tcp6       0      0 :::22                   :::*                    LISTEN      -                   
tcp6       0      0 :::80                   :::*                    LISTEN      -             
```      

Where do I configure the listen port so I can access it on my local lan? 

I'm more of a fedora user than a debian user, but I'm generally familiar with Linux.

**Comment by srchild** [comment]
"(_Not all processes could be identified, non-owned process info will not be shown, you would have to be root to see it all._)"

Try 

sudo netstat -ntap|grep LISTEN

**Comment by gosssamer** [comment]
Oh, yes, of course, oops.

However, there simply isn't anything running on ipv4 port 80. I can use lynx locally to connect on port 80, but not from a browser on the same local network.

It appears "caddy" is what's used to configure the listening port, based on it currently running on an ipv6 port 80?

**Comment by gosssamer** [comment]
Here's what netstat looks like as root:

```
# netstat -ntap|grep LISTEN
tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      413/python3         
tcp        0      0 127.0.0.1:8888          0.0.0.0:*               LISTEN      482/gotty           
tcp        0      0 127.0.0.1:8501          0.0.0.0:*               LISTEN      415/python3         
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      640/sshd: /usr/sbin 
tcp        0      0 0.0.0.0:21              0.0.0.0:*               LISTEN      908/inetd           
tcp        0      0 127.0.0.1:2019          0.0.0.0:*               LISTEN      903/caddy           
tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN      411/gotty           
tcp        0      0 0.0.0.0:8000            0.0.0.0:*               LISTEN      934/icecast2        
tcp6       0      0 :::80                   :::*                    LISTEN      903/caddy           
tcp6       0      0 :::22                   :::*                    LISTEN      640/sshd: /usr/sbin 
tcp6       0      0 ::1:8501                :::*                    LISTEN      415/python3         
```

Here's the journalctl output from restarting caddy:

```
# journalctl -l |grep caddy
...
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"info","ts":1694292167.2010887,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":""}
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"info","ts":1694292167.2109582,"logger":"admin","msg":"admin endpoint started","address":"localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"warn","ts":1694292167.211717,"logger":"http.auto_https","msg":"server is listening only on the HTTP port, so no automatic HTTPS will be applied to this server","server_name":"srv0","http_port":80}
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"info","ts":1694292167.2140155,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x400059a880"}
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"info","ts":1694292167.2177706,"logger":"http.log","msg":"server running","name":"srv0","protocols":["h1","h2","h3"]}
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"info","ts":1694292167.2185729,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/var/lib/caddy/.local/share/caddy"}
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"info","ts":1694292167.2186832,"logger":"tls","msg":"finished cleaning storage units"}
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"info","ts":1694292167.2186906,"msg":"autosaved config (load with --resume flag)","file":"/var/lib/caddy/.config/caddy/autosave.json"}
Sep 09 21:42:47 birdnet caddy[22550]: {"level":"info","ts":1694292167.2188501,"msg":"serving initial configuration"}
```

**Comment by srchild** [comment]
Mine looks pretty similar to yours, no obvious listener on port 80 IPV4

> birdnetpi2:~/BirdNET-Pi$ sudo netstat -ntap|grep LISTEN
> tcp        0      0 127.0.0.1:8001          0.0.0.0:*               LISTEN      789/icecast2
> tcp        0      0 127.0.0.1:8000          0.0.0.0:*               LISTEN      789/icecast2
> tcp        0      0 127.0.0.1:2019          0.0.0.0:*               LISTEN      775/caddy
> tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN      390/gotty
> tcp        0      0 0.0.0.0:21              0.0.0.0:*               LISTEN      779/inetd
> tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      595/sshd: /usr/sbin
> tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      392/python3
> tcp        0      0 127.0.0.1:8888          0.0.0.0:*               LISTEN      412/gotty
> tcp        0      0 127.0.0.1:8501          0.0.0.0:*               LISTEN      394/python3
> tcp6       0      0 :::22                   :::*                    LISTEN      595/sshd: /usr/sbin
> tcp6       0      0 :::80                   :::*                    LISTEN      775/caddy
> tcp6       0      0 ::1:8501                :::*                    LISTEN      394/python3

I have no problem accessing my Birdnet-Pi webpage on port 80 at http://birdnetpi.local/ from my Windows desktop using Chrome (or other browser)

**Reply by gosssamer** [reply]
It must then be accessing it over ipv6 in your case. My Linux box on my local net must not have ipv6 configured.

I'm accessing it like:

http://192.168.1.32/

From the local box:

```
# ifconfig                                                                                                                                        
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.1.32  netmask 255.255.255.0  broadcast 192.168.1.255
        inet6 fe80::6107:2d99:5d38:96c0  prefixlen 64  scopeid 0x20<link>
        ether e4:5f:01:fb:db:26  txqueuelen 1000  (Ethernet)
        RX packets 15520  bytes 1698979 (1.6 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 6830  bytes 1711317 (1.6 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

**Comment by gosssamer** [comment]
Okay, accessing it in an incognito browser "fixed" the problem, ugh. Apparently despite accessing it expliclty using http:// it still resorted to accessing it as https://.

I'm still curious how to explicitly set the ipv4 IP for this and/or why it isn't being displayed in netstat output?
